server/services/store/storetests/blocks.go

```diff
@@ -310,3 +317,128 @@ func testGetRootID(t *testing.T, store store.Store, tearDown func()) {
 		require.Error(t, err)
 	})
 }
+
+func testGetParentID(t *testing.T, store store.Store) {
+	userID := "user-id"
+
+	blocks, err := store.GetAllBlocks()
+	require.NoError(t, err)
+	initialCount := len(blocks)
+
+	blocksToInsert := []model.Block{
+		{
+			ID:         "parent",
+			RootID:     "parent",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "child1",
+			RootID:     "parent",
+			ParentID:   "parent",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "child2",
+			RootID:     "parent",
+			ParentID:   "parent",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "grandchild1",
+			RootID:     "parent",
+			ParentID:   "child1",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "grandchild2",
+			RootID:     "parent",
+			ParentID:   "child2",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "greatgrandchild1",
+			RootID:     "parent",
+			ParentID:   "grandchild1",
+			ModifiedBy: userID,
+		},
+	}
+
+	InsertBlocks(t, store, blocksToInsert)
+
+	blocks, err = store.GetAllBlocks()
+	require.NoError(t, err)
+	require.Len(t, blocks, initialCount+6)
+
+	t.Run("from root id", func(t *testing.T) {
+		parentID, err := store.GetParentID("parent")
+		require.NoError(t, err)
+		require.Equal(t, "", parentID)
+	})
+
+	t.Run("from child id", func(t *testing.T) {
+		parentID, err := store.GetParentID("grandchild1")
+		require.NoError(t, err)
+		require.Equal(t, "child1", parentID)
+	})
+
+	t.Run("from not existing id", func(t *testing.T) {
+		_, err := store.GetParentID("not-exists")
+		require.Error(t, err)
+	})
+}
+
+func testDeleteBlock(t *testing.T, store store.Store) {
+	userID := "user-id"
+
+	blocks, err := store.GetAllBlocks()
+	require.NoError(t, err)
+	initialCount := len(blocks)
+
+	blocksToInsert := []model.Block{
+		{
+			ID:         "block1",
+			RootID:     "block1",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "block2",
+			RootID:     "block2",
+			ModifiedBy: userID,
+		},
+		{
+			ID:         "block3",
+			RootID:     "block3",
+			ModifiedBy: userID,
+		},
+	}
+	InsertBlocks(t, store, blocksToInsert)
+
+	blocks, err = store.GetAllBlocks()
+	require.NoError(t, err)
+	require.Len(t, blocks, initialCount+3)
+
+	t.Run("exiting id", func(t *testing.T) {
+		// Wait for not colliding the ID+insert_at key
+		time.Sleep(1 * time.Millisecond)
+		err := store.DeleteBlock("block1", userID)
+		require.NoError(t, err)
+	})
+
+	t.Run("exiting id multiple times", func(t *testing.T) {
+		// Wait for not colliding the ID+insert_at key
+		time.Sleep(1 * time.Millisecond)
+		err := store.DeleteBlock("block1", userID)
+		require.NoError(t, err)
+		// Wait for not colliding the ID+insert_at key
+		time.Sleep(1 * time.Millisecond)
+		err = store.DeleteBlock("block1", userID)
+		require.NoError(t, err)
```

Comment:
@chenilim Is this the expected behavior?

Reply:
That's the current expected behavior - you can delete a block multiple times, since delete is an insert.

Off the top of my head, I think the current behavior is ok, optimizing for fast updates and deletes. Have to think of the downstream implications more.